server/rootless.go

```diff
@@ -17,7 +21,41 @@ func hasNetworkNamespace(config *rspec.Spec) bool {
 }
 
 func makeOCIConfigurationRootless(g *generate.Generator) {
-	g.Config.Linux.Resources = nil
+	// Resource limitations requires cgroup v2 delegation (https://rootlesscontaine.rs/getting-started/common/cgroup2/).
+	if g.Config.Linux.Resources != nil {
+		// cannot control device eBPF with rootless
+		g.Config.Linux.Resources.Devices = nil
+		v2Controllers := getAvailableV2Controllers()
+		if _, ok := v2Controllers["memory"]; !ok && g.Config.Linux.Resources.Memory != nil {
```

Comment:
Makes sense to reverse the order of checks:
```suggestion
		if r.Memory != nil && _, ok := v2Controllers["memory"]; !ok {
```
as the `!= nil` check is faster than a map lookup.

Reply:
That syntax is uncompilable, and benefit seems negligible